torchtext/vocab.py

```diff
@@ -92,57 +85,28 @@ def __eq__(self, other):
     def __len__(self):
         return len(self.itos)
 
-    def load_vectors(self, vectors, unk_init=torch.Tensor.zero_, expand_vocab=False):
+    def extend(self, v, sort=False):
+        words = sorted(v.itos) if sort else v.itos
+        for w in words:
+            if w not in self.stoi:
+                self.itos.append(w)
+                self.stoi[w] = len(self.itos) - 1
+
+    def load_vectors(self, vectors):
         """Arguments:
-              vectors: one of the available pretrained vectors or a list with each
-                  element one of the available pretrained vectors:
-                       charngram.100d
-                       fasttext.en.300d
-                       fasttext.simple.300d
-                       glove.42B.300d
-                       glove.840B.300d
-                       glove.twitter.27B.25d
-                       glove.twitter.27B.50d
-                       glove.twitter.27B.100d
-                       glove.twitter.27B.200d
-                       glove.6B.50d
-                       glove.6B.100d
-                       glove.6B.200d
-                       glove.6B.300d
-              unk_init (callback): by default, initialize out-of-vocabulary word vectors
-                  to zero vectors; can be any function that takes in a Tensor and
-                  returns a Tensor of the same size. Default: torch.Tensor.zero_
-              expand_vocab (bool): expand vocabulary to include all words for which
-                  the specified pretrained word vectors are available
+              vectors: one of or a list containing instantiations of the
+                  GloVe, CharNGram, or Vectors classes
```

Comment:
Not allowing these strings might be too breaking

Reply:
I don't think we should support both. The string wrapper was too brittle and forced everyone into GloVe format. It would also make it hard to deal with custom vectors defined by strings.

Reply:
All I mean is a dictionary that maps the existing strings to lambdas (thunks) that call the corresponding class constructors